bin/list-nginx-class-ingresses.rb

```diff
@@ -3,42 +3,42 @@
 # Script to build the list of ingresses and its namespaces with
 # the ingress-class annotations as "nginx" or empty i.e "" or ingress
 # with no annotations and into a json file
+# This script will list ingress for all non-prod namespaces. For production ingresses
+# change if non_production?(namespace) to if production?(namespace)
 
 require "json"
 require "open3"
 
 NGINX_CLASS_INGRESS_LIST_FILE = "nginx_class_ingresses.json"
 
 def main
-  nginx_class_ingresses = list_ingresses_has_annotations("nginx")
-  null_class_ingresses = list_ingresses_has_annotations("")
-  no_annotation_ingresses = list_ingresses_no_annotations
-  list = nginx_class_ingresses + null_class_ingresses + no_annotation_ingresses
+  $namespaces = get_namespaces
+  nginx_class_ingresses = ingresses_matching_class("nginx")
+  null_class_ingresses = ingresses_matching_class("")
+  no_annotation_ingresses = ingresses_matching_no_class
 
-  File.write(NGINX_CLASS_INGRESS_LIST_FILE, list.to_json)
+  list = nginx_class_ingresses.compact + null_class_ingresses.compact + no_annotation_ingresses.compact
+
+  puts "Total ingress listed: #{list.size}"
+
+  File.write(NGINX_CLASS_INGRESS_LIST_FILE, JSON.pretty_generate(list))
 end
 
-def list_ingresses_has_annotations(target_ingress_class)
+def ingresses_matching_class(target_ingress_class)
```

Comment:
You don't need the `ingress_array` variable here, because `map` returns an array anyway.

So you should be able to do this:
```
def ingresses_matching_class(target_ingress_class)
  get_ingresses
    .filter { |ingress| ingress.dig("metadata", "annotations", "kubernetes.io/ingress.class").to_s == target_ingress_class }
    .map { |ingress| non_production_tuple(ingress) }
end
```
Please double-check that that works correctly, but I'm fairly sure it's right.

The same technique will work for `ingresses_matching_no_class` as well.

Reply:
@digitalronin , if I dont do the ingress_array.push, I dont get the array returned. I think it is because the non_production_tuple returns a list and it needs to be build to an array.